Shared/Samples/Style symbols from mobile style file/StyleSymbolsFromMobileStyleFileView.SymbolOptionsListView.swift

```diff
@@ -0,0 +1,121 @@
+        
+        /// The view model for the sample.
+        @ObservedObject var model: Model
+        
+        /// The list of all the symbols and their associated data from the symbol style.
+        @State private var symbols: [SymbolDetails] = []
+        
+        var body: some View {
+            VStack {
+                Image(uiImage: model.currentSymbol?.image ?? UIImage())
+                
+                List {
+                    // Create a picker for each symbol category.
+                    ForEach(SymbolCategory.allCases, id: \.self) { category in
+                        Section(category.label) {
+                            Picker("Symbol Names", selection: $model.symbolOptionSelections.categoryKeys[category]) {
+                                ForEach(symbols.filter { $0.category == category.rawValue }, id: \.key) { symbol in
+                                    Label {
+                                        Text(symbol.label)
+                                    } icon: {
+                                        Image(uiImage: symbol.image)
+                                    }
+                                    .tag(symbol.key as String?)
+                                }
+                            }
+                            .pickerStyle(.inline)
+                            .labelsHidden()
+                        }
+                    }
+                    
+                    Section {
+                        ColorPicker("Color", selection: $model.symbolOptionSelections.color)
+                        VStack {
+                            HStack {
+                                Text("Size")
+                                Spacer()
+                                Text(model.symbolOptionSelections.size.formatted(.number.rounded()))
+                                    .foregroundColor(.secondary)
+                            }
+                            Slider(value: $model.symbolOptionSelections.size, in: 20...60, step: 1)
+                        }
+                    }
+                }
+            }
+            .task(id: model.symbolOptionSelections) {
+                // Update the current symbol when an option selection changes.
+                await model.updateCurrentSymbol(displayScale: displayScale)
+            }
+            .task(id: displayScale) {
+                // Update the symbols when the display scale changes.
+                symbols = await getSymbols(displayScale: displayScale)
+                await model.updateCurrentSymbol(displayScale: displayScale)
+            }
+        }
+        
+        /// Gets the symbols from the symbol style.
+        /// - Parameter displayScale: The display scale of the environment for creating symbol swatches.
+        /// - Returns:  An `Array` of `SymbolDetails` which contain a symbol and its associated information.
+        private func getSymbols(displayScale: Double) async -> [SymbolDetails] {
```

Comment:
if any of the `try?` calls fail in this function then the error information is discarded. You might consider making this function `throws` and doing something with the error. That would also simplify a lot of the code in this function, alleviating the need for some guard statements.

Reply:
I originally had the function throw, but I chose to go with `try?` so that if an error is throw in the task group, the function returns the symbols it could load. That way, you still get a list of symbols even if you have a faulty key, which I found is pretty easy to mess up. Also, I tried to mimic [a similar pattern](https://github.com/Esri/arcgis-maps-sdk-swift-samples/blob/main/Shared/Samples/Display%20content%20of%20utility%20network%20container/DisplayContentOfUtilityNetworkContainerView.Model.swift#L120-L135) I found in other samples. If you think that it would be better to have it throw I can definitely do that here and in `updateCurrentSymbol(displayScale:)`, though.

Reply:
I don't feel strongly about it. In general I don't think it's great to eat errors and I wouldn't want our users copying code that does that. But that's more of a samples app philosophical discussion that is probably out of scope of this PR.